### PR TITLE
use match-scanner library

### PR DESCRIPTION
**Pull request by noboruma**

### CODE REVIEW

File: pkg/scan/scanner.go [hunk 1]
```diff
@@ -1,25 +1,118 @@
...
+		}
+	}()
+
+	genscan.ApplyScan(context.Background(), extract, func(f extractor.ExtractedFile) {
+		if isExecutable(f.Filename) {
+			return
+		}
+		if isSharedLibrary(f.Filename) {
+			return
+		}
```
Comment (ibreakthecloud) [inline pkg/scan/scanner.go:106]:
```suggestion
		if isExecutable(f.Filename) || isSharedLibrary(f.Filename) {
			return
		}
```

File: pkg/scan/scanner.go [hunk 2]
```diff
@@ -1,25 +1,118 @@
...
+		buf := m[i][:0]
+		err := ScanFile(s, f.Filename, f.Content, f.ContentSize, &buf, "")
+		if err != nil {
+			log.Printf("err: %v", err)
```
Comment (ibreakthecloud) [inline pkg/scan/scanner.go:110]: todo:
 - use logrus and avoid printing in stdout 
 - also print path too, easier to debug

File: pkg/scan/process_image.go [hunk 3]
```diff
@@ -315,11 +293,7 @@ func ScanFile(s *Scanner, f *os.File, iocs *[]output.IOCFound, layer string) err
 	if len(matches) > 0 {
 		// output.PrintColoredIOC(tempIOCsFound, &isFirstIOC, fileMat.updatedScore, fileMat.updatedSeverity)
 		for _, m := range iocsFound {
-			if isSharedLib {
-				m.FileSeverity = "low"
-			} else {
-				m.FileSeverity = updatedSeverity
-			}
+			m.FileSeverity = updatedSeverity
```
[the remark this answers is not in the document]
Comment (noboruma) [reply pkg/scan/process_image.go:198]: @ibreakthecloud please propose an algorithm for the sev calculation, looks like you have some ideas in mind, let's review that in a separate PR

**Review by ibreakthecloud — APPROVED** [review]
we have good perf numbers regarding cpu and mem. 
my todos for later:
- fix the severity logic (have something better than what we have now)
- deepfence's own threatintel for malware, where we'll curate our own rules from multiple sources, this will result in more faster matching.

Currently rules are the only bottle necks we have.

**Pull request merged**